src/Sulu/Bundle/MediaBundle/Collection/Manager/CollectionManager.php

```diff
@@ -532,17 +532,16 @@ protected function getPreview($id, $locale)
                 foreach ($file->getFileVersions() as $fileVersion) {
                     if ($fileVersion->getVersion() == $file->getVersion()) {
                         $format = $this->getPreviewsFromFileVersion($media->getId(), $fileVersion, $locale);
+                        
                         if (!empty($format)) {
-                            $media = $format;
+                            return $format;
                         }
-                        break;
                     }
                 }
-                break;
             }
         }
 
-        return $media;
+        return null;
```

Comment:
`$media` is a "running" var and should not be returned ... so i have changed the return statement in the loop and return null here!